src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1415,10 +1415,194 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
         }
     }
 
+    bool       isScalar = false;
+    genTreeOps oper     = node->HWOperGet(&isScalar);
+
+    switch (oper)
+    {
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+        {
+            if (!comp->canUseEvexEncoding())
+            {
+                break;
+            }
+
+            GenTree* op1 = node->Op(1);
+            GenTree* op2 = node->Op(2);
+
+            LIR::Use use;
+            if (BlockRange().TryGetUse(node, &use))
+            {
+                // search for structure like:
+                /*
+                        /- A
+                        +- B
+                    t1 = binary logical op1
```

Comment:
Just to make  sure I follow the example in the comment - It is trying to find op1 and op2 has nested opertions basically.

Reply:
This is finding a bitwise operation that contains a nested bitwise operation so they can become a single ternary operation